pkg/cloudclient/gcp/private.go

```diff
@@ -115,10 +115,28 @@ func (c *Client) setDefaultAPIPublic(ctx context.Context, kclient client.Client,
 	return nil
 }
 
-func (c *Client) ensureDNSForService(kclient client.Client, svc *corev1.Service, dnsName string) error {
+func (c *GCPClient) ensureDNSForService(kclient client.Client, svc *corev1.Service, dnsName string) error {
 	// google.golang.org/api/dns/v1.Service is a struct, not an interface, which
 	// will make this all but impossible to write unit tests for
 
+	// Forwarding rule is necessary for rh-api lb setup
+	// Check forwarding rule exists first
+	if svc.Spec.Type == "LoadBalancer" {
```

Comment:
Is there ever a case where this this service won't be LoadBalancer? If so, should there be an error?

Reply:
I'm checking for the type to be on the safer side, and not throwing an error, because I don't want to interfere with any existing functionality for those services.

Reply:
The apischeme controller creates this Service, so we can rely on the code used to create the service there. If there's a type check, then each type should be explicitly handled.

Reply:
I'm not sure about other types and it seems too out of scope, so I'll remove this check and I can create a separate card for handling types here.

Reply:
done